### PR TITLE
Opt out of article count when the user removes consent

### PR DESCRIPTION
**Pull request by TomPretty**
## What does this change?
Opt users out of article count epics/banners if we don't have TCFv2 consent, as per [this card](https://trello.com/c/5QeO8WKw/2313-do-not-show-article-count-to-users-who-have-not-consented-to-the-cmp)

## Why?
We need to be compliant!

### CODE REVIEW

File: src/web/lib/contributions.ts [hunk 1]
```diff
@@ -112,5 +117,35 @@ export const shouldHideSupportMessaging = (
     isRecurringContributor(isSignedIn) ||
     isRecentOneOffContributor();
 
+const REQUIRED_CONSENTS_FOR_ARTICLE_COUNT = [1, 3, 7];
```
Comment (tomrf1) [inline src/web/lib/contributions.ts:120]: not the fault of this PR, but this is quite cryptic!

**Review by tomrf1 — APPROVED** [review]
great work!

File: src/web/lib/contributions.ts [hunk 2]
```diff
@@ -112,5 +117,35 @@ export const shouldHideSupportMessaging = (
...
+        onConsentChange(({ ccpa, tcfv2 }) => {
+            if (ccpa) {
+                resolve(true);
+            } else if (tcfv2) {
+                const hasRequiredConsents = REQUIRED_CONSENTS_FOR_ARTICLE_COUNT.every(
+                    (consent) => tcfv2.consents[consent],
+                );
+
+                if (!hasRequiredConsents) {
+                    removeArticleCountsFromLocalStorage();
```
Comment (tomrf1) [inline src/web/lib/contributions.ts:144]: 👍

File: src/web/lib/contributions.ts (hunk 1 above)
Comment (TomPretty) [reply src/web/lib/contributions.ts:120]: it really is!

**Pull request merged**